### PR TITLE
[Merged by Bors] - feat(widget): position argument with insert_text

### PR DESCRIPTION
**Pull request by EdAyers**
Adds support for specifying the position in the insert_text widget effect.

ref: https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/paste.20the.20goal.20at.20the.20cursor

### CODE REVIEW

**Comment by EdAyers** [comment]
Hi what needs to be done for this to get merged?

**Review by bryangingechen — APPROVED** [review]
Can you update https://github.com/leanprover-community/lean/blob/master/doc/widget_server.md as well? 

The code changes look good to me, but I still haven't studied the widget framework in detail so I'd like to wait for @gebner or someone else to chime in.

File: library/init/meta/widget/basic.lean [hunk 1]
```diff
@@ -172,10 +172,19 @@ inductive mouse_event_kind
 | on_mouse_enter
 | on_mouse_leave
 
-/-- An effect is an action at the root of the widget component hierarchy
-and can give instructions to the editor to perform some task. -/
+/-- An effect is some change that the widget makes outside of its own state.
+Usually, giving instructions to the editor to perform some task.
+- `insert_text_relative` will insert at a line relative to the position of the widget.
```
Comment (eric-wieser) [inline library/init/meta/widget/basic.lean:177]: What does "the position of the widget" mean?

File: library/init/meta/widget/basic.lean [hunk 2]
```diff
@@ -172,10 +172,19 @@ inductive mouse_event_kind
...
+- `reveal_position` will move the editor to view the given position.
+- `highlight_position` will add a text highlight to the given position.
+- `clear_highlighting` will remove all highlights created with `highlight_position`.
+- `copy_text` will copy the given text to the clipboard.
+- `custom` can be used to pass custom effects to the client without having to recompile Lean.
+-/
 meta inductive effect : Type
-| insert_text (text : string)
+| insert_text_absolute (file_name : option string) (p : pos) (text : string)
+| insert_text_relative (relative_line : int) (text : string)
```
Comment (eric-wieser) [inline library/init/meta/widget/basic.lean:187]: Does it make sense to include `relative_col` too?

**Comment by gebner** [comment]
> Hi what needs to be done for this to get merged?

Sorry, this flew completely under my radar.  LGTM.

File: library/init/meta/widget/basic.lean (hunk 2 above)
Comment (gebner) [reply library/init/meta/widget/basic.lean:187]: AFAIU it inserts a whole line, with matching indentation.  So column wouldn't make much sense.

**Comment by bryangingechen** [comment]
@EdAyers do you mind creating a followup PR which updates the widget-server doc and also clarifies the doc string that @eric-wieser pointed out?

**Pull request closed without merging**